oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionContext.java

```diff
@@ -152,6 +155,7 @@ public SessionContext(
         this.valueFactory = new ValueFactoryImpl(
                 delegate.getRoot(), namePathMapper, this.blobAccessProvider);
         this.sessionQuerySettings = sessionQuerySettings;
+        this.cacheMap = new HashMap<>();
```

Comment:
hm.... not sure how that works now... you would need to keep it for every node you take a look at which will result for the cache to get really big. that doesn't sound good to me.
on the other hand: storing a boolean for each node might also come with some overhead as one would have to invalidate it upon save/refresh.